cyberdrop_dl/scraper/crawlers/tiktok_crawler.py

```diff
@@ -0,0 +1,119 @@
+            filename, ext = get_filename_and_ext(image_url.name)
+            scrape_item.add_children()
+            await self.handle_file(image_url, new_scrape_item, filename, ext)
+        if self.manager.parsed_args.cli_only_args.download_tiktok_audios:
+            await self.handle_audio(new_scrape_item, post, False)
+
+    @error_handling_wrapper
+    async def profile(self, scrape_item: ScrapeItem) -> None:
+        scrape_item.set_type(FILE_HOST_PROFILE, self.manager)
+        async for json_data in self.profile_post_pager(scrape_item):
+            for item in json_data["data"]["videos"]:
+                post_url = URL(item["play"])
+                if item.get("images"):
+                    await self.handle_image_post(scrape_item, item)
+                elif not post_url.path.endswith("mp3"):
+                    filename, ext = f'{item["video_id"]}.mp4', "mp4"
+                    new_scrape_item = self.create_scrape_item(
+                        scrape_item, post_url, "", True, scrape_item.album_id, item["create_time"]
+                    )
+                    scrape_item.add_children()
+                    if self.manager.parsed_args.cli_only_args.download_tiktok_audios:
+                        await self.handle_audio(new_scrape_item, item)
+                    await self.handle_file(post_url, new_scrape_item, filename, ext)
+
+    @error_handling_wrapper
+    async def video(self, scrape_item: ScrapeItem) -> None:
+        async with self.request_limiter:
+            video_data_url = self.api_url.with_query({"url": str(scrape_item.url)})
+            json_data = await self.client.get_json(self.primary_base_domain, video_data_url, origin=scrape_item)
+
+            if scrape_item.album_id is None:
+                scrape_item.album_id = json_data["data"]["author"]["id"]
+                scrape_item.add_to_parent_title(self.create_title(scrape_item.url.parts[1][1:]))
+
+            json_data["data"]["video_id"] = json_data["data"].get("id")
+            if json_data["data"].get("images"):
+                await self.handle_image_post(scrape_item, json_data["data"])
+                return
+
+            video_url = URL(json_data["data"]["play"])
+            filename, ext = f'{json_data["data"]["video_id"]}.mp4', "mp4"
+            new_scrape_item = self.create_scrape_item(
+                scrape_item, video_url, "", True, scrape_item.album_id, json_data["data"]["create_time"]
+            )
+            scrape_item.add_children()
+            if self.manager.parsed_args.cli_only_args.download_tiktok_audios:
+                await self.handle_audio(new_scrape_item, json_data["data"])
+            await self.handle_file(video_url, new_scrape_item, filename, ext)
+
+    @error_handling_wrapper
+    async def handle_audio(self, scrape_item: ScrapeItem, data: dict, new_folder: bool = True) -> None:
+        audio_url = URL(data["music_info"]["play"])
+        filename = f'{data["music_info"]["title"]}.mp3'
```

Comment:
Title could potentially have invalid characters

```suggestion
        filename = f'{data["music_info"]["title"]}.mp3'
        filename, ext = get_filename_and_ext(filename)
```